src/client_builder.rs

```diff
@@ -141,10 +307,62 @@ where
         let tls = handshake(self.domain.as_ref(), tcp)?;
 
         let mut client = Client::new(tls);
-        if !self.starttls {
+        if self.mode != ConnectionMode::StartTls {
             client.read_greeting()?;
         }
 
         Ok(client)
     }
+
+    #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+    fn build_tls_connection(&self, tcp: TcpStream) -> Result<Connection> {
+        match self.tls_kind {
+            #[cfg(feature = "native-tls")]
+            TlsKind::Native => self.build_tls_native(tcp),
+            #[cfg(feature = "rustls-tls")]
+            TlsKind::Rust => self.build_tls_rustls(tcp),
+            TlsKind::Any => self.build_tls_any(tcp),
+        }
+    }
+
+    #[cfg(feature = "native-tls")]
+    fn build_tls_any(&self, tcp: TcpStream) -> Result<Connection> {
+        self.build_tls_native(tcp)
+    }
```

Comment:
How do you feel about making rustls the default instead of native-tls?

Reply:
I have no real opinion (other than I HAD to use native_tls for the project I am on as rustls wouldn't connect to 'older' machines)